### PR TITLE
fix: support 1s interval, add binance request rate limiter

### PR DESCRIPTION
**Pull request by zenixls2**

### CODE REVIEW

File: pkg/bbgo/session.go [hunk 1]
```diff
@@ -497,12 +501,13 @@ func (session *ExchangeSession) Positions() map[string]*types.Position {
...
 	s, ok = session.marketDataStores[symbol]
-	if !ok {
+	// FIXME: the returned MarketDataStore when !ok will be empty
+	/*if !ok {
 		s = NewMarketDataStore(symbol)
 		s.BindStream(session.MarketDataStream)
 		session.marketDataStores[symbol] = s
 		return s, true
-	}
+	}*/
```
Comment (c9s) [inline pkg/bbgo/session.go:510]: why not initialize a new market data store when it's not found?

File: pkg/bbgo/activeorderbook.go [hunk 2]
```diff
@@ -92,60 +92,50 @@ func (b *ActiveOrderBook) waitAllClear(ctx context.Context, waitTime, timeout ti
...
 // GracefulCancel cancels the active orders gracefully
-func (b *ActiveOrderBook) GracefulCancel(ctx context.Context, ex types.Exchange) error {
+func (b *ActiveOrderBook) GracefulCancel(ctx context.Context, ex types.Exchange, orders ...types.Order) error {
+	// if no orders are given, set to cancelAll
+	if len(orders) == 0 {
+		orders = b.Orders()
+	} else {
+		// simple check on given input
+		for _, o := range orders {
+			if o.Symbol != b.Symbol {
```
Comment (c9s) [inline pkg/bbgo/activeorderbook.go:132]: make sure `b.Symbol != ""`

File: pkg/bbgo/activeorderbook.go [hunk 3]
```diff
@@ -168,6 +156,10 @@ func (b *ActiveOrderBook) GracefulCancel(ctx context.Context, ex types.Exchange)
 		}
 
 		log.Debugf("[ActiveOrderBook] waiting %s for %s orders to be cancelled...", waitTime, b.Symbol)
+		for _, o := range b.Orders() {
+			b.Remove(o)
+		}
+		break
```
Comment (c9s) [inline pkg/bbgo/activeorderbook.go:162]: why break here?

File: pkg/bbgo/activeorderbook.go [hunk 4]
```diff
@@ -92,60 +92,50 @@ func (b *ActiveOrderBook) waitAllClear(ctx context.Context, waitTime, timeout ti
 	}
 }
 
-// Cancel cancels the given order from activeOrderBook gracefully
-func (b *ActiveOrderBook) Cancel(ctx context.Context, ex types.Exchange, order types.Order) error {
-	if !b.Exists(order) {
-		return fmt.Errorf("cannot find %v in orderbook", order)
+func (b *ActiveOrderBook) CancelNoWait(ctx context.Context, ex types.Exchange, orders ...types.Order) error {
```
Comment (c9s) [inline pkg/bbgo/activeorderbook.go:95]: can you convert the NoWait  to an option defined in ActiveOrderBook? like activeOrders.SetNoWait(true)

Comment (zenixls2) [reply pkg/bbgo/activeorderbook.go:95]: we might need both operations on the same activeOrderbook in a single strategy.  
for example, in emergency exit, we could like all the orders to be successfully cancelled, so we call GracefulCancel, while in high frequency trading, we still need CancelNoWait.

File: pkg/bbgo/activeorderbook.go (hunk 3 above)
Comment (zenixls2) [reply pkg/bbgo/activeorderbook.go:162]: testing code not yet been cleaned up

File: pkg/exchange/binance/parse.go [hunk 5]
```diff
@@ -350,6 +350,11 @@ func parseWebSocketEvent(message []byte) (interface{}, error) {
 		err = json.Unmarshal([]byte(message), &event)
 		return &event, err
 
+	case "aggTrade":
```
Comment (c9s) [inline pkg/exchange/binance/parse.go:353]: can we have aggTrade in one single PR?

**Comment by c9s** [comment]
conflict~

**Pull request closed without merging**